docs/data/sdk-middleware.md

```diff
@@ -0,0 +1,130 @@
+| `event.user_id` | string |
+| `event.device_id` | string |
+| `event.user_properties` | { [key: string]: any } |
+| `extra` | { [x: string]: any } |
+
+For other platforms, middleware can access and modify the entire Event JSON object, allowing for comprehensive adjustments as needed. Learn more at [here](../../../analytics/apis/http-v2-api/#keys-for-the-event-argument).
+
+## Middleware examples
+
+Use an Middleware to modify event properties, transformation, filtering, routing to third-party destinations, and more:
+!!!example
+
+    ???code-example "Drop-event middleware example(click to expand)"
+
+        ```ts
+         amplitude.addEventMiddleware((payload, next) => {
+          const {eventType} =  payload.event;
+          if (shouldSendEvent(eventType)) {
+            next(payload)
+          } else {
+            // event will not continue to following middleware or be sent to Amplitude
+            console.log(`Filtered event: ${eventType}`);
+          }
+        });
+        ```
+        
+    ???code-example "Remove PII (Personally Identifiable Information) (click to expand)"
+
+        ```js
+        amplitude.addEventMiddleware((payload, next) => {
+          const { event } = payload;
+          if (hasPii(event.event_properties)) {
+            obfuscate(payload.event.event_properties);
+          }
+          next(payload);
+        });
+        ```
+    
+    ???code-example "Enrich Event Properties (click to expand)"
+
+        ```js
+        ampli.client.addEventMiddleware((payload, next) => {
+          const { event } = payload;
+          if (needsDeviceId(event)) {
+            payload.event.deviceId = getDeviceId();
+          }
+          next(payload)
+        });
+        ```
+
+    ???code-example "Forward data to other services, but not Amplitude (click to expand)"
+
+        ```js
+        import amplitude from 'amplitude/sdk'
+        import adroll from 'adroll';
+        import segment from 'segment';
+        import snowplow from 'snowplow';
+        
+        amplitude.addEventMiddleware((payload, next) => {
+          const { event, extra } = payload;
+          segment.track(event.event_type, event.event_properties, { extra.anonymousId })
+          adroll.track();
+          snowplow.track(event.event_type, event.event_properties, extra.snowplow.context);
+          // next();
+        });
+        ```
+
+    ???code-example "Use client-side validation (click to expand)"
+
+        ```js
+        amplitude.addEventMiddleware((payload, next) => {
+          if (isDevelopment && !SchemaValidator.isValid(payload.event)) {
+            throw Error(`Invalid event ${event.event_type}`);
+          }
+          next(payload);
+        });
+        
+        amplitude.addEventMiddleware((payload, next) => {
+          const { event, extra } = payload;
+          segment.track(event.event_type, event.event_properties, { extra.segment.anonymousId })
+          next(payload);
```

Comment:
Is this a separate example?

Reply:
Yes, this is an example of forwarding events to other platforms.

Reply:
I think this should not be mixed with the client side validation example. Can you please update so each example is focused on a single feature.